code/modules/research/bepis.dm

```diff
@@ -0,0 +1,225 @@
+		dropturf = drop_location()
+	gauss_major = (gaussian(major_threshold, std) - negative_cash_offset)	//This is the randomized profit value that this experiment has to surpass to unlock a tech.
+	gauss_minor = (gaussian(minor_threshold, std) - negative_cash_offset)	//And this is the threshold to instead get a minor prize.
+	gauss_real = (gaussian(banked_cash, std*inaccuracy_percentage) + positive_cash_offset)	//this is the randomized profit value that your experiment expects to give.
+	say("Real: [gauss_real]. Minor: [gauss_minor]. Major: [gauss_major].")
+	if(gauss_real >= gauss_major) //Major Success.
+		say("Experiment concluded with major success. New technology node discovered on technology disc.")
+		banked_cash = 0
+		new /obj/item/disk/tech_disk/major(dropturf,1)
+		flick("chamber_flash",src)
+		update_icon_state()
+		return
+	else if(gauss_real >= gauss_minor) //Minor Success.
+		var/reward_number = 1
+		say("Experiment concluded with partial success. Dispensing compiled research efforts.")
+		reward_number = rand(1,2)
+		if(reward_number == 1)
+			new /obj/item/stack/circuit_stack/full(dropturf)
+		if(reward_number == 2)
+			new /obj/item/airlock_painter/decal(dropturf)
+		banked_cash = 0
+		flick("chamber_flash",src)
+		update_icon_state()
+		return
+	else if(gauss_real <= -1)	//Critical Failure
+		say("ERROR: CRITICAL MACHIME MALFUNCTI- ON. CURRENCY IS NOT CRASH. CANNOT COMPUTE COMMAND: 'make bucks'") //not a typo, for once.
+		banked_cash = 0
+		new /mob/living/simple_animal/deer(dropturf, 1)
+		use_power(500000 * power_saver) //To prevent gambling at low cost and also prevent spamming for infinite deer.
+		flick("chamber_flash",src)
+		update_icon_state()
+		return
+	else	//Minor Failure
+		error_cause = pick("attempted to sell grey products to American dominated market.","attempted to sell gray products to British dominated market.","placed wild assumption that PDAs would go out of style.","simulated product #76 damaged brand reputation mortally.","simulated business model resembled 'pyramid scheme' by 98.7%.","product accidently granted override access to all station doors.")
+		say("Experiment concluded with zero product viability. Cause of error: [error_cause]")
+		banked_cash = 0
+		flick("chamber_flash",src)
+		update_icon_state()
+	return
+
+/obj/machinery/rnd/bepis/update_icon_state()
+	if(is_operational() && (powered == TRUE))
+		chamber_status = 1
+	if(((powered == FALSE) && (banked_cash == 0)) || (!is_operational()))
+		chamber_status = 0
+	if((powered == TRUE) && (banked_cash > 0) && (is_operational()))
+		chamber_status = 2
+	if (((powered == FALSE) && (banked_cash > 0)) || (banked_cash > 0) && (!is_operational()))
+		chamber_status = 3
+	if(panel_open == TRUE)
+		chamber_status = 4
+	switch(chamber_status)
+		if(0)
+			icon_state = "chamber"
+		if(1)
+			icon_state = "chamber_active"
+		if(2)
+			icon_state = "chamber_active_loaded"
+		if(3)
+			icon_state = "chamber_loaded"
+		if(4)
+			icon_state = "chamber_open"
```

Comment:
Invert this, set icon_state directly and early return. i.e.
```
/obj/machinery/rnd/bepis/update_icon_state()
	if(panel_open == TRUE)
		icon_state = "chamber_open"
		return
	if (((powered == FALSE) && (banked_cash > 0)) || (banked_cash > 0) && (!is_operational()))
		icon_state = "chamber_loaded"
		return
```
and so on. Then get rid of chamber_status.
Still feels a little messy, you are checking some things again and again, but its a little better.